src/erlcloud_aws.erl

```diff
@@ -73,23 +73,23 @@
 
 aws_request_xml(Method, Host, Path, Params, #aws_config{} = Config) ->
     Body = aws_request(Method, Host, Path, Params, Config),
```

Comment:
It looks like `aws_request` [raises](https://github.com/erlcloud/erlcloud/blob/e5f03228189b7582d6e825c605089bd4ee652f3b/src/erlcloud_aws.erl#L116) `erlang:error({aws_error, Reason})`. I think `aws_request_xml` should remain consistent to that.

Reply:
Updated to raise an exception in `aws_request` path.  `aws_request2` and `aws_request4` will return error tuples to the calling modules.